chunky/src/java/se/llbit/chunky/renderer/scene/PathTracer.java

```diff
@@ -166,6 +165,22 @@ public static boolean pathTrace(Scene scene, Ray ray, WorkerState state, int add
               ray.emittance.z = ray.color.z * ray.color.z *
                   currentMat.emittance * scene.emitterIntensity;
               hit = true;
+            } else if(scene.emittersEnabled) {
+              // Sample emitter
+              boolean sampleOne = ray.depth != 0;
+              if(sampleOne) {
+                Grid.EmitterPosition pos = scene.getEmitterGrid().sampleEmitterPosition((int) ray.o.x, (int) ray.o.y, (int) ray.o.z, random);
+                if(pos != null) {
+                  // TODO Sampling a random point on the model would be better than using a random point in the middle of the cube
```

Comment:
This could be done once all models are made out of quads, i.e. when the json block loader is finished one day